### PR TITLE
Add availability and consistency docs.

### PR DESCRIPTION
**Pull request by gianm**
Describes transactional ingestion and atomic replacement. Also, this patch
deletes some bad advice from the javadocs for SegmentTransactionalInsertAction.

### CODE REVIEW

File: docs/design/architecture.md [hunk 1]
```diff
@@ -213,6 +213,75 @@ this will generally start off `true` and then become `false` as the segment is p
...
+
+On the **ingestion side**, Druid's primary [ingestion methods](../ingestion/index.md#ingestion-methods) are all
+pull-based and offer transactional guarantees. This means that you are guaranteed that ingestion using these will
+publish in an all-or-nothing manner:
+
+- Supervised "seekable-stream" ingestion methods like [Kafka](../development/extensions-core/kafka-ingestion.md) and
+[Kinesis](../development/extensions-core/kinesis-ingestion.md). With these methods, Druid commits stream offsets to its
+[metadata store](#metadata-storage) alongside segment metadata, in the same transaction. Note that ingestion of data
+that has not yet been published can be rolled back if ingestion tasks fail. In this case, partially-ingested data
+discarded, and Druid will resume ingestion from the last committed set of stream offsets. This ensures exactly-once
```
Comment (surekhasaharan) [inline docs/design/architecture.md:230]: partially-ingested data discarded -> partially-ingested data "is" discarded

Comment (gianm) [reply docs/design/architecture.md:230]: Thanks, fixed.

**Comment by maytasm** [comment]
@gianm @surekhasaharan Travis docs job are now failing due to this change. https://travis-ci.org/github/apache/druid/jobs/705996918

**Comment by jihoonson** [comment]
@maytasm thank you for the catch. I raised https://github.com/apache/druid/pull/10152.

**Comment by gianm** [comment]
@surekhasaharan, it looks like the Travis CI build was still in progress when you merged this patch. Please wait for the tests to finish running before merging things.

**Pull request merged**